ci-operator/config/jianzhangbjz/learn-operator/jianzhangbjz-learn-operator-master.yaml

```diff
@@ -0,0 +1,76 @@
+base_images:
+  base:
+    name: "4.7"
+    namespace: ocp
+    tag: base
+binary_build_commands: make build
+build_root:
+  image_stream_tag:
+    name: release
+    namespace: openshift
+    tag: golang-1.14
+images:
+- dockerfile_path: build/Dockerfile.ci
+  from: base
+  inputs:
+    bin:
+      paths:
+      - destination_dir: .
+        source_path: /tmp/learn-operator
+      - destination_dir: .
+        source_path: /tmp/bin
+  to: learn-operator
+operator:
+  bundles:
+  - context_dir: manifests
+    dockerfile_path: bundle.Dockerfile
+  substitutions:
+  - pullspec: quay.io/olmqe/learn-operator:v1
+    with: learn-operator
+promotion:
+  name: learn
+  namespace: ocp
```

Comment:
OK, I'll say this bluntly. You have no idea what you are doing. You are not part of `ocp`. You should not promote to `ocp` namespace without understanding the consequences. Just remove `promotion`. When you actually get an idea of why you might need it, you can add it again, with some sensible configuratiojn.

Reply:
Actually, I'm familiar with the `BuildConfig` and `ImageStream`. But, I not very clear about the `name`, `namespace` description here: https://docs.ci.openshift.org/docs/architecture/ci-operator/#publishing-container-images I thought the `name` and `namespace` just for creating an `ImageStreamTag` called `namespace/name`. Must I set the `additional_images` and `excluded_images` for the `promotion`? Thanks!

Reply:
Just remove `promotion`

Reply:
You actually understand the meaning correctly - the `name` and `namespace` specify the imagestream into which the images will be promoted after merge. But again, you are *not* a part of `ocp`, so you should *not* promote into `ocp` namespace.

Reply:
> the stable:component tag, published as ocp/4.5:mycomponent containing the output component itself

And, for the description above, I couldn't find the `component`/`mycomponent` in that example:
```console
promotion:
  additional_images:
    repo-scripts: "`src`"    # promotes "src" as "repo-scripts"
  excluded_images:
  - "mytests" # does not promote the test image
  namespace: "ocp"
  name: "4.5"
```

Reply:
Anyway, I just promote it to my own namespace.

Reply:
> And, for the description above, I couldn't find the component/mycomponent in that example:

They are the names of images built in `images` stanza, defined in an example above. And again, the examples show principles (what is possible to be configured), not the exact values to be uses (the values are determined by the specifics of each component or repository). Promoting to an own namespace sounds good.

Reply:
Thanks! I also want to add more explanation for this doc and example too.